### PR TITLE
[SPARK-3340] Deprecate ADD_JARS and ADD_FILES

### PR DESCRIPTION
**Pull request by azagrebin**
I created a patch that disables the environment variables.
Thereby scala or python shell log a warning message to notify user about the deprecation 
with the following message:
scala: "ADD_JARS environment variable is deprecated, use --jar spark submit argument instead"
python: "Warning: ADD_FILES environment variable is deprecated, use --py-files argument instead"

Is it what is expected or the code associated with the variables should be just completely removed?
Should it be somewhere documented?

### CODE REVIEW

**Comment by srowen** [comment]
ok to test

**Comment by andrewor14** [comment]
@azagrebin thanks for working on this. However, deprecated just means it will be released in a future released, but not in the current release. If we log a warning saying that they are deprecated we still need to support them, at least for another release (until 1.4 or something).

**Comment by azagrebin** [comment]
Hi @andrewor14, thanks for the comment. So the variables need to be supported until some next release and a warning should be logged that they are deprecated (may be not supported in some next release), right? I have done another commit.

File: repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala [hunk 1]
```diff
@@ -1064,15 +1064,18 @@ class SparkILoop(
...
 
   def getAddedJars: Array[String] = {
     val envJars = sys.env.get("ADD_JARS")
-    val propJars = sys.props.get("spark.jars").flatMap { p =>
-      if (p == "") None else Some(p)
+    envJars match {
+      case Some(_) =>
+        logWarning("ADD_JARS environment variable is deprecated, use --jar spark submit argument instead")
+      case _ =>
     }
```
Comment (andrewor14) [inline repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala]: minor point but I think it's more readable here if you just do
```
if (envJars.isDefined) {
  logWarning(...)
}
```

File: repl/scala-2.11/src/main/scala/org/apache/spark/repl/Main.scala [hunk 2]
```diff
@@ -51,6 +51,11 @@ object Main extends Logging {
 
   def getAddedJars: Array[String] = {
     val envJars = sys.env.get("ADD_JARS")
+    envJars match {
+      case Some(_) =>
+        logWarning("ADD_JARS environment variable is deprecated, use --jar spark submit argument instead")
+      case _ =>
+    }
```
Comment (andrewor14) [inline repl/scala-2.11/src/main/scala/org/apache/spark/repl/Main.scala]: same here

**Comment by azagrebin** [comment]
yeah, really better, thanks for the hint, committed again

File: repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala [hunk 3]
```diff
@@ -1064,15 +1064,15 @@ class SparkILoop(
...
   implicit def loopToInterpreter(repl: SparkILoop): SparkIMain = repl.intp
   private def echo(msg: String) = Console println msg
 
   def getAddedJars: Array[String] = {
     val envJars = sys.env.get("ADD_JARS")
-    val propJars = sys.props.get("spark.jars").flatMap { p =>
-      if (p == "") None else Some(p)
-    }
+    if (envJars.isDefined)
+      logWarning("ADD_JARS environment variable is deprecated, use --jar spark submit argument instead")
```
Comment (andrewor14) [inline repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala]: our style guide actually requires us to put `{ } around the content of an `if`. I'll do this myself when I merge don't worry.

**Comment by andrewor14** [comment]
LGTM I'm merging this into master and 1.3 thanks

**Comment by andrewor14** [comment]
Hey @azagrebin what is your JIRA name? I would like to give you credit for this patch.

**Comment by azagrebin** [comment]
@andrewor14, thanks for the brackets and the credit, I have actually read the style guide but forgot them, sorry for that. I have also created the JIRA account, my username is the same: azagrebin.

**Pull request closed without merging**